### DISCUSSION TITLE
[Feature] Idea to retain facial recognition info after running 'Detect All Faces'

### DISCUSSION THREAD

**Discussion by radh21301** [immich-app/immich#12473] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, running the ' FACE DETECTION' job on all faces removes all the information the user added - Names and DOB of people. This is a tedious job to do again.

**Solution? - Maybe a bit hacky?**
This might sound hacky, but could possible be made more elegant?

Say we have a curated list of images associated with the name and DOB of a person, like the following:
* /library/{user}/{name of person}/text file or something with the DOB (this file can be used for other information in the future as well.)
  * If this is not a good idea, the DOB can be stored with the name of the person.
    * Like /library/{user}/{name of person} - DOB:2022-11-21
    * /library/admin/Jack - DOB:2022-11-21
* Under each folder for that person, can be a collection of curated images which contain that person. Note: These images should contain only one face - the face of that person.
* These folders need not be generated by the user - a normal user would not want to go into the filesystem to hack around like this. They can be generated by Immich when they run facial recognition for the first time, and they add names and DOB to the faces. Immich then generated these folders with this information.
* Similarly, they need not add images to the folders..the picture they add as the profile picture for that person, can be automatically added to the folder.
* Now, when a user reruns facial recognition - Immich can user the images in these curated folders in the cluster as well. If an cluster contains these curated images, then the information for that face can be automatically populated.

Example:
* /library/admin/Jack - DOB:2022-11-21
  *  IMG_2165.jpg
* /library/admin/Penny - DOB:2021-02-11
  *  IMG_4365.jpg
.
.
.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#12335 adds the capability to refresh faces without clearing existing data.

**Reply by radh21301** [reply]
Awesome.
I just changed my facial recognition model and ran this on all assets! Wish I saw this PR sooner! :)
